openfisca_aotearoa/variables/acts/citizenship/citizenship.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+
+from openfisca_core.model_api import Variable
+from openfisca_core.periods import DAY, ETERNITY
+from openfisca_aotearoa.entities import Person
+
+
+class citizenship__citizenship_by_grant_may_be_authorized(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = DAY
+    reference = "http://www.legislation.govt.nz/act/public/1977/0061/latest/DLM443855.html"
+
+    def formula_2005_04_20(persons, period, parameters):
+
+        return (persons('age', period) >= parameters(period).citizenship.by_grant.minimum_age_threshold) * \
+            persons('is_of_full_capacity', period) * \
+            persons('citizenship__meets_minimum_presence_requirements', period) * \
+            persons('citizenship__is_of_good_character', period) * \
+            persons('citizenship__has_sufficient_knowledge_of_the_responsibilities_and_privileges_attaching_to_nz_citizenship', period) * \
+            persons('citizenship__has_sufficient_knowledge_of_the_english_language', period) * \
+            (persons('citizenship__intends_to_reside_in_nz', period) + persons('citizenship__intends_to_enter_or_continue_crown_service', period))
+
+
+class citizenship__meets_minimum_presence_requirements(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = DAY
+    label = u"Applicant was present in New Zealand for a min of 1,350 days during the 5 years immediately preceding the date of application"
+    reference = "http://www.legislation.govt.nz/act/public/1977/0061/latest/DLM443855.html"
+
+    def formula(persons, period, parameters):
+        # persons('immigration__entitled_to_stay_indefinitely', period) * \
+        # (ii) for at least 240 days in each of those 5 years,—
+        # being days during which the applicant was entitled in terms of the Immigration Act 2009 to be in New Zealand indefinitely
+        # for p in [period.offset(offset) for offset in range(-365, 1)]:
+        return persons('citizenship__meets_5_year_presence_requirement', period) * \
+            persons('citizenship_meets_each_year_minimum_presence_requirements', period)
```

Comment:
@Br3nda I think you intended to have `citizenship__meets_each_year_minimum_presence_requirements`. Missing extra underscore

Reply:
fixed